src/abi/map.ts

```diff
@@ -353,7 +353,7 @@ export const MapFunctions: ReadonlyMap<string, AbiFunction> = new Map([
                 const [self] = args;
                 checkMapType(self, ref);
 
-                return { kind: "ref", name: "Cell", optional: true };
+                return { kind: "ref", name: "Cell", optional: false };
```

Comment:
so, this wouldn't fail at runtime?

```tact
contract Foo {
    get fun foo(): Slice {
        let x: map<Int, Int> = emptyMap();
        return x.asCell().asSlice();
    }
}
```

Reply:
If I understand correctly, now we just generate the qualifier expression as is, so `mp.asCell()` becomes `mp` in the FunC code:

https://github.com/tact-lang/tact/blob/7b1dd5463113bfe9b678c145a0d7507eacfd288c/src/abi/map.ts#L374

Thus, `asCell` simply returns non optional `Cell` type